src/jenkins.py

```diff
@@ -245,30 +250,114 @@ def _install_configs(container: ops.Container) -> None:
         container.push(JCASC_CONFIG_FILE_PATH, jenkins_casc_config_file, user=USER, group=GROUP)
 
 
-def _install_plugins(container: ops.Container) -> None:
+def _get_groovy_proxy_args(proxy_config: state.ProxyConfig) -> typing.Iterable[str]:
+    """Get proxy arguments for proxy configuration Groovy script.
+
+    Args:
+        proxy_config: The proxy settings to apply.
+
+    Yields:
+        Groovy script proxy arguments.
+    """
+    if proxy_config.https_proxy:
+        yield f"'{proxy_config.https_proxy.host}'",
+        yield f"{proxy_config.https_proxy.port}",
+        yield f"'{proxy_config.https_proxy.user or ''}'",
+        yield f"'{proxy_config.https_proxy.password or ''}'",
+    else:
+        # http proxy and https proxy value cannot both be None since proxy_config would be parsed
+        # as None.
+        proxy_config.http_proxy = typing.cast(HttpUrl, proxy_config.http_proxy)
+        yield f"'{proxy_config.http_proxy.host}'",
+        yield f"{proxy_config.http_proxy.port}",
+        yield f"'{proxy_config.http_proxy.user or ''}'",
+        yield f"'{proxy_config.http_proxy.password or ''}'",
+    if proxy_config.no_proxy:
+        yield f"'{proxy_config.no_proxy}'"
+
+
+def _configure_proxy(
+    container: ops.Container,
+    proxy_config: state.ProxyConfig | None = None,
+    client: jenkinsapi.jenkins.Jenkins | None = None,
+) -> None:
+    """Configure Jenkins proxy settings if proxy configuration values are provided.
+
+    Args:
+        container: The Jenkins workload container
+        proxy_config: The proxy settings to apply.
+        client: The API client used to communicate with the Jenkins server.
+
+    Raises:
+        JenkinsProxyError: if an error occurred running proxy configuration script.
+    """
+    if not proxy_config:
+        return
+
+    client = client if client is not None else _get_client(get_admin_credentials(container))
+    parsed_args = ", ".join(_get_groovy_proxy_args(proxy_config))
+    try:
+        client.run_groovy_script(f"proxy = new ProxyConfiguration({parsed_args})\nproxy.save()")
+    except jenkinsapi.custom_exceptions.JenkinsAPIException as exc:
+        logger.error("Failed to configure proxy, %s", exc)
+        raise JenkinsProxyError("Proxy configuration failed.") from exc
+
+
+def _get_java_proxy_args(proxy_config: state.ProxyConfig) -> typing.Iterable[str]:
+    """Get JVM system property arguments for proxy.
+
+    Args:
+        proxy_config: The proxy settings to apply.
+
+    Yields:
+        JVM System property proxy arguments.
+    """
+    if proxy_config.http_proxy:
+        yield f"-Dhttp.proxyHost={proxy_config.http_proxy.host}",
+        yield f"-Dhttp.proxyPort={proxy_config.http_proxy.port}",
+        if proxy_config.http_proxy.user and proxy_config.http_proxy.password:
+            yield f"-Dhttp.proxyUser={proxy_config.http_proxy.user}",
```

Comment:
I think the trailing comma needs to be deleted

Reply:
Yup, just realized that, sorry about that!